pkg/virtualization/core/virtualsystem/virtualmachine.go

```diff
@@ -883,6 +892,49 @@ func (vm *VirtualMachine) GetResourceAllocationSettingData(rtype v2.ResourcePool
 	return
 }
 
+func (vm *VirtualMachine) GetResourceAllocationSettingDataBySubType(resourceSubType virtconstant.ResourceSubType) (col *v2.CIM_ResourceAllocationSettingData, err error) {
+	settings, err := vm.GetVirtualSystemSettingData()
+	if err != nil {
+		return
+	}
+	defer settings.Close()
+
+	rasdcol, err := settings.GetAllRelated("CIM_ResourceAllocationSettingData")
+	if err != nil {
+		return
+	}
+	defer rasdcol.Close()
+
+	for _, ins := range rasdcol {
+		rasd, err1 := v2.NewCIM_ResourceAllocationSettingDataEx1(ins)
+		if err1 != nil {
+			err = err1
+			return
+		}
+
+		sourceResourceSubType, err1 := rasd.GetProperty("ResourceSubType")
+		if err1 != nil || sourceResourceSubType == nil {
+			continue
+		}
+
+		if string(resourceSubType) == sourceResourceSubType {
+			instance, err1 := rasd.Clone()
+			if err1 != nil {
+				err = err1
+				return
+			}
+			col, err1 = v2.NewCIM_ResourceAllocationSettingDataEx1(instance)
+			if err1 != nil {
+				err = err1
```

Comment:
It is unlikely to fail here, but if it fails, we would leak the newly cloned instance. Can you close that instance here, on failure